### PR TITLE
safeguard for multisig actor

### PR DESCRIPTION
**Pull request by waynewyang**
Some safeguard for getTransaction and ChangeRequireMent.

### CODE REVIEW

File: chain/actors/actor_multisig.go [hunk 1]
```diff
@@ -210,6 +210,9 @@ func (msa MultiSigActor) Approve(act *types.Actor, vmctx types.VMContext,
 	}
 
 	tx := self.getTransaction(params.TxID)
+	if tx == nil {
+		return nil, aerrors.Wrap(err, "could not get transaction")
```
Comment (whyrusleeping) [inline chain/actors/actor_multisig.go]: aerrors.Wrap is wrong here, the err youre passing it isnt related to the problem. Instead use `Newf` and give it an exit code.

**Comment by waynewyang** [comment]
@whyrusleeping   I've update the PR, but I have a question, how should I define the `RetCode`, and where is the Spec about `RetCode`?

**Comment by waynewyang** [comment]
> @whyrusleeping I've updated the PR, but I have a question, how should I define the `RetCode`, and where is the Spec about `RetCode`?

**Comment by whyrusleeping** [comment]
For the retcodes, just pick a value that isnt used yet (so they are easy to distinguish). This has not made it into the spec yet, so lotus is the spec in this regard.

**Comment by waynewyang** [comment]
Got it, thanks

File: chain/actors/actor_multisig.go [hunk 2]
```diff
@@ -29,11 +29,11 @@ func (msas MultiSigActorState) isSigner(addr address.Address) bool {
 	return false
 }
 
-func (msas MultiSigActorState) getTransaction(txid uint64) *MTransaction {
-	if txid > uint64(len(msas.Transactions)) {
-		return nil
+func (msas MultiSigActorState) getTransaction(txid uint64) (*MTransaction, ActorError) {
+	if 0 == len(msas.Transactions) || txid > uint64(len(msas.Transactions)) {
```
Comment (magik6k) [inline chain/actors/actor_multisig.go]: That's better, but still not entirely correct, should be:
```suggestion
	if txid >= uint64(len(msas.Transactions)) {
```

Comment (whyrusleeping) [reply chain/actors/actor_multisig.go]: oh whoops, it was wrong in master D:

**Comment by waynewyang** [comment]
@whyrusleeping My email is wayne.yang@arsyun.com , can you invate me to the #fil-lotus-dev channel,thanks~

**Comment by waynewyang** [comment]
Hi @whyrusleeping I'm back now， I've join #fil-lotus-dev and my handle is also `waynewyang`.

**Pull request closed without merging**